bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java

```diff
@@ -0,0 +1,193 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.boschshc.internal.devices.bridge;
+
+import static org.eclipse.jetty.http.HttpMethod.POST;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Consumer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.openhab.binding.boschshc.internal.devices.bridge.dto.LongPollError;
+import org.openhab.binding.boschshc.internal.devices.bridge.dto.LongPollResult;
+import org.openhab.binding.boschshc.internal.devices.bridge.dto.SubscribeResult;
+import org.openhab.binding.boschshc.internal.exceptions.BoschSHCException;
+import org.openhab.binding.boschshc.internal.exceptions.LongPollingFailedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Handles the long polling to the Smart Home Controller.
+ * 
+ * @author Christian Oeing - Initial contribution
+ */
+@NonNullByDefault
+public class LongPolling {
+    @SuppressWarnings("serial")
+    private class AbortLongPolling extends BoschSHCException {
+    }
+
+    private final Logger logger = LoggerFactory.getLogger(LongPolling.class);
+
+    private final Gson gson = new Gson();
+
+    private final ScheduledExecutorService scheduler;
+
+    /**
+     * Handler for long poll results.
+     */
+    private final Consumer<LongPollResult> handleResult;
+
+    /**
+     * Handler for unrecoverable.
+     */
+    private final Consumer<Throwable> handleFailure;
+
+    /**
+     * Current running long polling request.
+     */
+    private @Nullable Request request;
+
+    /**
+     * Indicates if long polling was aborted.
+     */
+    private boolean aborted = false;
+
+    public LongPolling(ScheduledExecutorService scheduler, Consumer<LongPollResult> handleResult,
+            Consumer<Throwable> handleFailure) {
+        this.scheduler = scheduler;
+        this.handleResult = handleResult;
+        this.handleFailure = handleFailure;
+    }
+
+    public void start(BoschHttpClient httpClient) throws LongPollingFailedException {
+        // Subscribe to state updates.
+        String subscriptionId = this.subscribe(httpClient);
+        this.executeLongPoll(httpClient, subscriptionId);
+    }
+
+    public void stop() {
+        // Abort long polling.
+        this.aborted = true;
+        Request request = this.request;
+        if (request != null) {
+            request.abort(new AbortLongPolling());
+            this.request = null;
+        }
+    }
+
+    /**
+     * Subscribe to events and store the subscription ID needed for long polling
+     *
+     * Method is synchronous.
```

Comment:
is it?

Reply:
Yes, as far as I can see. Just in contrast to other methods of the class which are handled asynchronous, e.g. executeLongPoll. Should I remove the comment?

Reply:
Well saying that it is synchronous implies that only a single thread can call it at a time, but I see no indication of that in the code. I would suggest removing the comment.

Reply:
Alright, I see. I just saw it as the opposite of an asynchronous http request, but have not thought about threading. I will remove it to not confuse people :)

Reply:
Removed the comment.